client/rpc/debug/src/lib.rs

```diff
@@ -504,6 +504,9 @@ where
 			Ok(moonbeam_rpc_primitives_debug::Response::Block)
 		};
 
+		// Offset to account for old buggy transactions that are in trace not in the ethereum block
+		let mut tx_position_offset = 0;
```

Comment:
This should be something like: `traces.len().saturating_sub(eth_transactions_by_index.len())`.

An then below, we first need to filter all traces that have `tx_position` < `tx_position_offset`. Then map it and use `eth_transactions_by_index.get(&(trace.tx_position.saturating_sub(tx_position_offset)))` to get the hash.

Reply:
It's a typo, I wanted to write `trace.tx_position - tx_position_offset` instead of `+`